In [10]:
import pandas as pd


In [11]:
technew = pd.read_excel(r'data\Technology Skills.xlsx')     
techold = pd.read_excel(r'data\Technology Skills 260.xlsx') 
techdiff = pd.concat([technew, pd.concat([techold]*2)]).drop_duplicates(['Title','Commodity Code'], keep=False)
techdiff.shape


(386, 6)

We see there are 386 rows different between the 2 Technology Skills tables. Results don't tell us which are old vs new, unfortunately. 

In [12]:
unnew = pd.read_excel(r'data\UNSPSC Reference.xlsx')
unold = pd.read_excel(r'data\UNSPSC Reference 26.0.xlsx')
undiff = pd.concat([unnew, pd.concat([unold]*2)]).drop_duplicates(['Commodity Code', 'Commodity Title'], keep=False)
undiff.shape

(378, 8)

WE see there are 378 different rows between the 2 UNSPSC tables. 

In [13]:
toolsnew = pd.read_excel(r'data\Tools Used.xlsx')     
toolsold = pd.read_excel(r'data\Tools Used 26.0.xlsx') 
toolsdiff = pd.concat([toolsnew, pd.concat([toolsold]*2)]).drop_duplicates(['Title','Commodity Code'], keep=False)
toolsdiff.shape

(467, 5)

There are 467 different rows between the 2 tables of Tools Used. 

In [14]:
undiff.to_csv('output_UNSPSC_diff.csv')
techdiff.to_csv('output_Technology_Skills_diff.csv')
toolsdiff.to_csv('output_Tools_Used_diff.csv')